### ISSUE TITLE
Java regex pathmatcher missing.

### ISSUE THREAD

**Issue by dleacock** [akka/akka-http#2749]
Apologies up front if the error is due to myself. I'm unable to find any java side information on this online or on the lightbend forums.

Intent: I wanted to create a route that matches for *.ts  (i.e  localhost:8080/video/1.ts). I've tried pathSuffix for ".ts" which didn't work so I attempt to use regex. 

1. According to the [Basic PathMatchers](https://doc.akka.io/docs/akka-http/current/routing-dsl/path-matchers.html) there should be a PathMatchers.regex in the javadsl.  Every other PathMatcher exists in javadsl (slash, end, segment, remaining, etc) but no .regex

2. I tried to create my own PathMatcher1<String> based off a regex query and use 
` path(PathMatcher1<T> p, java.util.function.Function<T,Route> inner)` 

I constructed my pathmatcher through:
```
Pattern regexPattern = Pattern.compile('some regex');
PathMatcher1<String> pathMatcher= PathMatchers.segment(regexPattern);
```
and my function,
```
private Function<String, Route> regexFunctionRoute = new Function<String, Route>() {
       @Override
        public Route apply(String pathString) {
            return path(pathString, () ->
                    get(() -> extractRequest(request -> complete(testHandler.handleTS(request))))
            );
      }
    };
```
When I try to use path(pathMatcher, regexFunctionRoute) it doesn't recognize pathMatcher as PathMatcher1<T>.  Intellij tells me that 

`Cannot resolve method 'path(akka.http.javadsl.server.PathMatcher1<java.lang.String>, java.util.function.Function<java.lang.String, akka.http.javadsl.server.Route>'.  `

It recognizes the function as being correct, but not the PathMatcher1<String>.  Am I doing something wrong or is there something missing from the javadsl?

Thank you very much
David

**Comment by johanandren** [comment]
The docs mentioning `PathMatchers.regexp` is an oversight, it should simply be `PathMathers.segment(Pattern)`. Not sure why your code does not compile, in the PR to fix the docs I included a test case that uses a regex from Java that you can compare your code with: https://github.com/akka/akka-http/pull/2767/files#diff-ec67fe384341708da4960895aaec995dR293

**Comment by dleacock** [comment]
Thank you for your help.

**Comment by jrudolph** [comment]
Docs fixed in #2767.
